src/core/thread/mle_router.cpp

```diff
@@ -2085,23 +2085,33 @@ otError MleRouter::UpdateChildAddresses(const AddressRegistrationTlv &aTlv, Chil
 {
     const AddressRegistrationEntry *entry;
     Ip6::Address address;
+    Ip6::Address mleid;
     Lowpan::Context context;
     uint8_t count;
+    bool withMlEid = false;
     char stringBuffer[Ip6::Address::kIp6AddressStringSize];
+    otError error = OT_ERROR_NONE;
+
+    // Retrieve registered MlEid firstly
+    mleid.Clear();
+    aChild.GetIp6AddressWithPrefix(GetMeshLocal16(), OT_MESH_LOCAL_PREFIX_SIZE, mleid);
```

Comment:
Suggestion: Let's update this change after https://github.com/openthread/openthread/pull/2504 is merged.

Instead of `GetIp6AddressWithPrefix()` we can  maybe add a method on `Child` to give us the mesh-lcoal address directly since it'd be stored/saved differently after https://github.com/openthread/openthread/pull/2504... (e.g. something like `otError Child::GetMeshLocalIp6Address(Ip6::Address &aAddress)`.

Reply:
Agree with @abtink.

Reply:
@jwhui  just merged https://github.com/openthread/openthread/pull/2504. 
Just submitted https://github.com/openthread/openthread/pull/2511 which will add `Child::GetMeshLocalIp6Address()` to the `Child` class.

Reply:
Thanks. Will rebase.